### PR TITLE
Remove previous default tag on non-revisioned installation

### PR DESCRIPTION
**Pull request by Monkeyanator**
Part of the fix for #34932 

Installing Istio without a revision should be considered as installing a default revision, and therefore we should remove the previous default tag on install to avoid mutating webhook overlap issues (otherwise analyzer will detect duplicate between default injector and default tag and fail out).

Noticed another issue when looking into this: since we rely on the default tag to provide resource validation now, and the in-cluster operator does not use this, we have no validation when installing with in-cluster operator currently. Need to address this before 1.12 release (either document to use `istioctl tag set default` or do it in `istioctl operator init` similar to how we autogenerate tag for `istioctl install`).

- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: operator/cmd/mesh/install.go
Comment (howardjohn) [inline operator/cmd/mesh/install.go:189]: This introduces a (likely small) downtime that may lead to a pod not being injected. I think this is something that is never acceptable, even for a small time, as failiure to inject a pod can lead to authz policy bypass

**Review by howardjohn — CHANGES_REQUESTED** [review]
Overwrite sounds fine, but  delete+create is suspicious

File: operator/cmd/mesh/install.go
Comment (Monkeyanator) [reply operator/cmd/mesh/install.go:190]: Hm there is already a small window of injection downtime (between calling `DeactivateIstioInjectionWebhook` and creating the generated revision tag).... will have to rework this too.

Comment (howardjohn) [reply operator/cmd/mesh/install.go:190]: ok, I think that is worth doing but we can merge this in the mean time if desired as its not making the situation any worse I guess

Comment (Monkeyanator) [reply operator/cmd/mesh/install.go:190]: Yeah this is meant to unblock docs test update, I think we should either merge or revert the breaking PR

Comment (Monkeyanator) [reply operator/cmd/mesh/install.go:190]: Bit of a brain dump but another thing we're trying to avoid here is having a window of double-injection by different revisions.  For example, default tag points to revision `foo`, we install without a revision so then we also have the istio-sidecar-injector webhook injecting for default labels until at the end of the installation we deactivate that webhook.

Maybe this is taping over the fact that there shouldn't be an injector for `istio-injection=enabled` installed through the `istio-discovery` chart anymore now that we control this through default tag... would fix the issue of in-cluster operator reconciling over the deactivated webhook too

**Review by howardjohn — APPROVED** [review]

**Comment by Monkeyanator** [comment]
cc @istio/wg-user-experience-maintainers for review

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**
